core/execution/common/liquidity_provision.go

```diff
@@ -487,12 +487,13 @@ func (m *MarketLiquidity) AmendLiquidityProvision(
 		}
 	}
 
-	if !m.liquidityEngine.IsLiquidityProvider(party) {
+	existingAmendment := m.liquidityEngine.PendingProvisionByPartyID(party)
+	if !m.liquidityEngine.IsLiquidityProvider(party) && existingAmendment == nil {
```

Comment:
I think the `existingAmendment` is not needed since then `IsLiquidityProvider` now checks for pending as well correct?

Reply:
correct - what I wasn't sure of is if having IsLiquidityProvider account for pending has any unexpected negative consequences unrelated to here. you can leave it and remove after merging.